ravenframework/BaseClasses/BaseInterface.py

```diff
@@ -136,7 +136,7 @@ def removeMetaKeys(self, args):
 
   ################################
   # API (legacy) - these should go away as we convert existing systems
-  def readXML(self, xmlNode, variableGroups=None, globalAttributes=None):
+  def readXML(self, xmlNode, variableGroups=None, globalAttributes=None, **kwargs):
```

Comment:
I would strongly recommend that BaseInterface not add **kwargs here.  This means that readXML for every subclass can now pass any keyword argument without getting an error.

Reply:
I agree with you. The reason is because the readXML method in the code interface accepts WorkingDir as input also. I can change **kwargs to workingDir=None. Is it ok to you? otherwise, I need to overwrite the readXML method in code interface.

Reply:
workingDir=None is better than adding **kwargs.  overwriting the readXML method in code interface is also fine with me.

Reply:
I have updated the code with workingDir=None

Reply:
Thanks.